File: p002.hs
{- 
 - Solution to Project Euler problem 2
 - By Nayuki Minase
 - 
 - http://nayuki.eigenstate.org/page/project-euler-solutions
 - https://github.com/nayuki/Project-Euler-solutions
 -}


main = putStrLn (show ans)
ans = sum $ filter even $ takeWhile (<= 4000000) fibonacci

fibonacci = 1 : 2 : (zipWith (+) fibonacci (tail fibonacci))

File: p002.py
# 
# Solution to Project Euler problem 2
# by Project Nayuki
# 
# http://www.nayuki.io/page/project-euler-solutions
# https://github.com/nayuki/Project-Euler-solutions
# 


ans = 0
x = 1
y = 2
while x <= 4000000:
	if x % 2 == 0:
		ans += x
	x, y = y, x + y
print(ans)
